web/modules/custom/rdf_etl/src/Plugin/EtlAdms2ConvertPassPluginBase.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\rdf_etl\Plugin;
+
+use Drupal\Core\Database\Database;
+use Drupal\Core\Plugin\PluginBase;
+use Drupal\KernelTests\KernelTestBase;
+
+/**
+ * Provides a base class for Adms2ConvertPass plugins.
+ */
+abstract class EtlAdms2ConvertPassPluginBase extends PluginBase implements EtlAdms2ConvertPassInterface {
+
+  /**
+   * The SPARQL database connection.
+   *
+   * @var \Drupal\rdf_entity\Database\Driver\sparql\Connection
+   */
+  protected $sparql;
+
+  /**
+   * Constructs a Drupal\Component\Plugin\PluginBase object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   */
+  public function __construct(array $configuration, string $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->sparql = Database::getConnection('default', 'sparql_default');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function convert(array $data): void {}
+
+  /**
+   * {@inheritdoc}
+   */
+  public function performAssertions(KernelTestBase $test): void {}
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTestingRdfData(): ?string {
+    return NULL;
+  }
+
+  /**
+   * Gets all the triples from a given graph, ordered by subject.
```

Comment:
Keep in mind that there is a limit to the fetched triples which I think defaults to a thousand or ten thousands. 
You should perform checks with more than this amount of triples to ensure consistency (or at least that the query indeed fetches all results).

This number is set in the `virtuoso.ini`'s parameter `ResultSetMaxRows`. Maybe a test with that limit set to a small number can also avoid memory consuming tests.

Reply:
I prefered to make this query safe by splitting the results in chunks with `LIMIT` and `OFFSET` clauses. I went with a chunk of 10000 items which seems for me reasonable. The default value  of `ResultSetMaxRows` is one million and I cannot imagine a system being configured with less than 10000 triples.